esmvaltool/recipes/testing/recipe_deangelis15nat_fig1_fast.yml

```diff
@@ -0,0 +1,63 @@
+# ESMValTool
+# recipe_deangelis15nat.yml
+---
+documentation:
+  title: "Evaluate water vapor short wave radiance absorption schemes of ESMs with the observations"
+  description: |
+    Recipe for reproducing the plots in DeAngelis et al., 2015 Nature
+    Short version of Fig. 1b for test cases.
+
+  authors:
+    - weigel_katja
+
+  maintainer:
+    - weigel_katja
+
+  references:
+    - deangelis15nat
+
+  projects:
+    - eval4cmip
+
+preprocessors:
+  spatial_mean:
+    area_statistics:
+      operator: mean
+
+
+diagnostics:
+
+  deangelisf1b:
+    description: Bar plot similar to (DeAngelis Fig. 1b).
+    # Global multi-model experiment mean for flux variables, e.g. rlnst, rsnst, lvp, hfss
+    variables:
+      rlnst: &spatial_mean_cmip5_r1i1p1_amon_t2ms
+        preprocessor: spatial_mean
+        project: CMIP5
+        ensemble: r1i1p1
+        mip: Amon
+        field: T2Ms
+        derive: true
+        force_derivation: false
+      rsnst:
+        <<: *spatial_mean_cmip5_r1i1p1_amon_t2ms
+      lvp:
+        <<: *spatial_mean_cmip5_r1i1p1_amon_t2ms
+      hfss:
+        <<: *spatial_mean_cmip5_r1i1p1_amon_t2ms
+        derive: false
+    additional_datasets:
+      - {dataset: GFDL-CM3, exp: piControl, start_year: 1,
+         end_year: 1}
+      - {dataset: GFDL-CM3, exp: rcp85, start_year: 2091,
+         end_year: 2091}
+      - {dataset: GFDL-CM3, exp: abrupt4xCO2, start_year: 141, end_year: 141}
+        # - {dataset: ACCESS1-0, exp: piControl, start_year: 300, end_year: 301}
+        # - {dataset: ACCESS1-0, exp: rcp85, start_year: 2091, end_year: 2092}
+        # - {dataset: ACCESS1-0, exp: abrupt4xCO2, start_year: 300, end_year: 301}
```

Comment:
Why are these commented out but not deleted?

Reply:
> Thanks! I noticed that there was no provenance for the figure, so I added it for the diagnostic used in this recipe (it looks like it's also missing for the other diagnostic scripts in the same folder though, maybe this would be something to look into).

The provenance should be there, but was so far written in work not in plot.
Should it be in plot now?

Reply:
This change:
    - fig.savefig(get_plot_filename('bar_all', cfg), dpi=300)
    + plot_file = get_plot_filename('bar_all', cfg)
    + fig.savefig(plot_file, dpi=300)
    
    Would add an extra variable name. My main issue with the code style checks is always, that it complains about too many variabel names -  therefore I don't like it.

Reply:
> Why are these commented out but not deleted?

Sorry, I forgot to delete it (I like the version with two models more, but it is not reliably under 5 minutes).

Reply:
> Should it be in plot now?

Provenance should be provided for all output files, so both for the files in 'work` and `plot`. This is also written in the documentation: https://docs.esmvaltool.org/en/latest/community/diagnostic.html#provenance-items-provided-by-the-diagnostic-script

Reply:
> I like the version with two models more, but it is not reliably under 5 minutes

Is there a lot of variation in the runtime? I think we don't need to bee too strict on the 5 minutes, especially now that we only have a few trsting recipes.

Reply:
> > Should it be in plot now?
> 
> Provenance should be provided for all output files, so both for the files in `work` and `plot`. This is also written in the documentation: https://docs.esmvaltool.org/en/latest/community/diagnostic.html#provenance-items-provided-by-the-diagnostic-script

In the doku it says:

> with ProvenanceLogger(cfg) as provenance_logger:
>       provenance_logger.log(diagnostic_file, provenance_record)
> The diagnostic_file can be obtained using [esmvaltool.diag_scripts.shared.get_diagnostic_filename](https://docs.esmvaltool.org/en/latest/api/esmvaltool.diag_scripts.shared.html#esmvaltool.diag_scripts.shared.get_diagnostic_filename).

This would write it in work. In the example which is linked there, it is probably written in plot, but it is hard to follow, because I'd first need to follow functions from shared (save_figure) which probably did not even exists, when diagnostic was written. 
That it is really hard to follow and a really good example of things, which are not really backward compatibel and make it hard for people to contribute to the Tool, because even such a small thing like shortening an existing diagnistic for testing purposes creates a lot of extra work :(

Reply:
> > I like the version with two models more, but it is not reliably under 5 minutes
> 
> Is there a lot of variation in the runtime? I think we don't need to bee too strict on the 5 minutes, especially now that we only have a few testing recipes. The last run by the esmvalbot only took 20 seconds.

Yes, but I think this could be a special feature of Levantes shared queue. With two models (and 3 experiments each) I got up to nearly 7 minutes run time.

Reply:
> > > Should it be in plot now?
> > 
> > 
> > Provenance should be provided for all output files, so both for the files in `work` and `plot`. This is also written in the documentation: https://docs.esmvaltool.org/en/latest/community/diagnostic.html#provenance-items-provided-by-the-diagnostic-script
> 
> In the doku it says:
> 
> > with ProvenanceLogger(cfg) as provenance_logger:
> > provenance_logger.log(diagnostic_file, provenance_record)
> > The diagnostic_file can be obtained using [esmvaltool.diag_scripts.shared.get_diagnostic_filename](https://docs.esmvaltool.org/en/latest/api/esmvaltool.diag_scripts.shared.html#esmvaltool.diag_scripts.shared.get_diagnostic_filename).
> 
> This would write it in work. In the example which is linked there, it is probably written in plot, but it is hard to follow, because I'd first need to follow functions from shared (save_figure) which probably did not even exists, when diagnostic was written. That it is really hard to follow and a really good example of things, which are not really backward compatible and make it hard for people to contribute to the Tool, because even such a small thing like shortening an existing diagnistic for testing purposes creates a lot of extra work :(

@bouweandela The version included now produces 8 files for one plot (*.png + 3 provenance files in plot, *.nc + 3 provenance files in work), while this is not really an issue for this small example, I think it its not a good idea for diagnostics with many figues. Some of the other deangelis diagnostics and some of the martin2018 diagnostics produce for example [Number of Variables] * [Number of Models + 1] plots (e.g. map plots for each model and the multi model mean for several variables). This can easily result in 50-100 plots, if there are 6 additional provenance files that are really too many (and from the code I'd assume the provenance files in work and plot are identical?).

Reply:
> > > I like the version with two models more, but it is not reliably under 5 minutes
> > 
> > 
> > Is there a lot of variation in the runtime? I think we don't need to bee too strict on the 5 minutes, especially now that we only have a few testing recipes. The last run by the esmvalbot only took 20 seconds.
> 
> Yes, but I think this could be a special feature of Levantes shared queue. With two models (and 3 experiments each) I got up to nearly 7 minutes run time.

I added the second model again, at the moment it is around 4 minutes for that on one CPU, shared queue on Levante  (but the times there really seem to vary).

Reply:
> This would write it in work. In the example which is linked there

Indeed, the example is for writing to work, but it also says: "For each output file produced by the diagnostic script, ..." and with that I meant that it's also needed for other files, but I guess that is not clear enough..